File: Database/Assembly_SRPG_JP/CustomTargetParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.CustomTargetParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

using System;

namespace SRPG
{
  public class CustomTargetParam
  {
    public string iname;
    public string[] units;
    public string[] jobs;
    public string[] unit_groups;
    public string[] job_groups;
    public string first_job;
    public string second_job;
    public string third_job;
    public ESex sex;
    public int birth_id;
    public int element;

    public bool Deserialize(JSON_CustomTargetParam json)
    {
      this.iname = json.iname;
      if (json.units != null)
      {
        this.units = new string[json.units.Length];
        for (int index = 0; index < json.units.Length; ++index)
          this.units[index] = json.units[index];
      }
      if (json.jobs != null)
      {
        this.jobs = new string[json.jobs.Length];
        for (int index = 0; index < json.jobs.Length; ++index)
          this.jobs[index] = json.jobs[index];
      }
      if (json.unit_groups != null)
      {
        this.unit_groups = new string[json.unit_groups.Length];
        for (int index = 0; index < json.unit_groups.Length; ++index)
          this.unit_groups[index] = json.unit_groups[index];
      }
      if (json.job_groups != null)
      {
        this.job_groups = new string[json.job_groups.Length];
        for (int index = 0; index < json.job_groups.Length; ++index)
          this.job_groups[index] = json.job_groups[index];
      }
      this.first_job = json.first_job;
      this.second_job = json.second_job;
      this.third_job = json.third_job;
      this.sex = (ESex) json.sex;
      this.birth_id = json.birth_id;
      string[] strArray = new string[6]
      {
        json.dark.ToString(),
        json.shine.ToString(),
        json.thunder.ToString(),
        json.wind.ToString(),
        json.water.ToString(),
        json.fire.ToString()
      };
      string empty = string.Empty;
      foreach (string str in strArray)
        empty += str;
      this.element = Convert.ToInt32(empty, 2);
      return true;
    }
  }
}

File: Database_V2/ParamFunctions_untested/CustomTargetParam.py
def CustomTargetParam(json):
    this={}#CustomTargetParamjson)
    if 'iname' in json:
        this['iname'] = json['iname']
    #if(json.units!=null)
        #for(intindex=0index<json.units.Length++index)
        if 'units' in json:
            this['units'] = json['units']
    #if(json.jobs!=null)
        #for(intindex=0index<json.jobs.Length++index)
        if 'jobs' in json:
            this['jobs'] = json['jobs']
    #if(json.unit_groups!=null)
        #for(intindex=0index<json.unit_groups.Length++index)
        if 'unit_groups' in json:
            this['unit_groups'] = json['unit_groups']
    #if(json.job_groups!=null)
        #for(intindex=0index<json.job_groups.Length++index)
        if 'job_groups' in json:
            this['job_groups'] = json['job_groups']
    if 'first_job' in json:
        this['first_job'] = json['first_job']
    if 'second_job' in json:
        this['second_job'] = json['second_job']
    if 'third_job' in json:
        this['third_job'] = json['third_job']
    if 'sex' in json:
        this['sex'] = ENUM['ESex'][json['sex']]
    if 'birth_id' in json:
        this['birth_id'] = json['birth_id']
    #stringstrArray=newstring[6]
        #json.dark.ToString(),
        #json.shine.ToString(),
        #json.thunder.ToString(),
        #json.wind.ToString(),
        #json.water.ToString(),
        #json.fire.ToString()
        #}
        #stringempty=string.Empty
        #foreach(stringstrinstrArray)
        #empty+=str
        #this.element=Convert.ToInt32(empty,2)
        #returntrue
return this
